### ISSUE TITLE
Setting a property with invalid json takes down system

### ISSUE THREAD

**Issue by ivakegg** [apache/accumulo#3909]
So I was updating the service executor pools configuration on a system and inadvertantly missed the final bracked on the property.  This results in "invalid json".  All of the tservers then started spamming, in what appeared to be a very tight loop, this error.  Many tserver subsequently ran out of memory and others were left in a bad state and became unresponsive (deemed unreachable but did not halt).  Also the monitor got pounded so hard with messages that the node itself became unresponsive.  Unfortunately the manager was running on the same node and became unreachable as well.

I was able to rectify the property using the property editor and after a while the system calmed down.  However I still had to drop the entire cluster to be able to recover all of the tservers that were in a bad state.

Basically, setting a json property with invalid json should not take down the system.

**Comment by dlmarion** [comment]
We likely need to add some logic to PropUtil.validateProperties for this. However, without annotating which properties accept json, we might have to hardcode the list for now.

**Comment by keith-turner** [comment]
> We likely need to add some logic to PropUtil.validateProperties for this. However, without annotating which properties accept json, we might have to hardcode the list for now.


It would be nice to validate these properties at set time, but not sure its possible.  Would need to overcome the following in validation

 * The compaction service config spans multiple properties.
 * Validating JSON would not be enough and could break plugins that are not the default.  A planner plugin implementation may not take json, or it could take json but expect a very specific schema.

Opened #3912 to make the system behave better when there is bad config as identified in this issue. Need to do some more general follow on work to address per table compaction dispatchers having config that makes them throw an exception, that may cause a similar log storm.

**Comment by dtspence** [comment]
@keith-turner @dlmarion As a possibly related issue (but less severe), we observed that an incorrect `accumulo.properties` configuration caused the default compaction planner to report an exception during initialization - but still allow the tablet-server to start-up (i.e. without a default service).

The error manifested in the tablet-server logs as (note we believe there is no default service due to the error at start-up):
```
ERROR: Tablet<> returned non-existent compaction service default for compaction type SYSTEM. Check the table compaction dispatcher configuration. Attempting to fall back to default service.
```

At tablet server start-up there is a message:
```
[compactions.CompactionManager] ERROR: Failed to create compaction service default with planner:<attempted configuration>
com.google.gson.JsonSyntaxException ...
...
at org.apache.accumulo.core.spi.compaction.DefaultCompactionPlanner.init(DefaultCompactionPlanner.java:54)
at org.apache.accumulo.tserver.compactions.CompactionService.<init>(CompactionService.java:107)
at org.apache.accumulo.tserver.compactions.CompactionManager.<init>(CompactionManager.java:200)
at org.apache.accumulo.tserver.TabletServer.run(TabletServer.java:774)
...
```

**Comment by ctubbsii** [comment]
Fixed with #3912 and further validation of JSON in general being partially added with #3927.
